autorest/error.go

```diff
@@ -28,6 +28,9 @@ type DetailedError struct {
 
 	// Message is the error message.
 	Message string
+
+	// Service Error is the response body of failed API.
+	ServiceError interface{}
```

Comment:
this is the interface to which response body was unmarshalled. I need only the bytes and can let the caller consume it and do the unmarshalling. If you are okay with that I can make that change.